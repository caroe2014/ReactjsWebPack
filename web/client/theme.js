// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import { css, injectGlobal } from 'styled-components';

const theme = {
  colors: {
    black: '#000000',
    navigationMenu: '#48a942',
    navigationMenuHover: 'cornflowerblue',
    primary: '#347CFF',
    primaryButton: '#347CFF',
    loginField: '#AAA7AC',
    placeHolder: '#909090',
    loginBoxColor: '#434149',
    open: '#48a942',
    close: 'red',
    footer: '#48a942',
    textGrey: '#505050',
    paymentSuccessTitle: '#505050',
    inputBottomBorder: 'lightgray',
    backgroundGrey: '#E6E6E6',
    text: 'black',
    primaryText: '#383838',
    secondaryText: '#a6a6a6',
    hint: '#a6a6a6',
    light: 'white',
    error: '#cc0000',
    tile: '#505050',
    addToCart: '#347CFF',
    itemTile: '#3C474C0',
    darkGrey: '#464646',
    itemBorder: 'lightgrey',
    addOnTab: '#347CFF',
    validationError: '#cc0000',
    loginBackground: '#303030',
    loginFormBackground: '#484848',
    disableBackground: '#E4E4E4',
    addonBg: '#EEEEEE',
    glutenFree: '#8d7356',
    spicy: '#ed1c24',
    vegan: '#0d4a22',
    vegetarian: '#0d4a22',
    new: '#ff6000',
    healthy: '#359919',
    contextBackground: '#f0f0f3',
    loader: '#ababab',
    disableButton: '#D1D1D1'
  },
  fontFamily: 'Roboto',
  receiptFontFamily: 'Roboto',
  fontSize: {
    xl: '2.4rem',
    lg: '1.8rem',
    md: '1.3rem',
    nm: '1rem',
    sm: '0.75rem'
  },
  fontSizeFromProps: (props) => {
    if (props.xl) return props.theme.fontSize.xl;
    if (props.lg) return props.theme.fontSize.lg;
    if (props.md) return props.theme.fontSize.md;
    if (props.nm) return props.theme.fontSize.nm;
    if (props.sm) return props.theme.fontSize.sm;
    return 'inherit';
  },
  spacing: {
    xl: '8em',
    lg: '4em',
    md: '2em',
    nm: '1em',
    sm: '0.5em',
    none: '0em'
  },
  mediaBreakpoints: {
    mobile: (...args) => css`@media (max-width: 32em) {${css(...args)}}`,
    tablet: (...args) => css`@media (max-width: 72em) {${css(...args)}}`,
    desktop: (...args) => css`@media (min-width: 72em) {${css(...args)}}`
  },
  shadow: '#cfcfcf 0.1px 0.1px 3px 0px',
  nav: {
    height: '70px'
  }
};

injectGlobal`
  .agilysys-icon-new {
    color: ${theme.colors.new}
  }
  .agilysys-icon-healthy {
    color: ${theme.colors.healthy}
  }
  .agilysys-icon-vegetarian {
    color: ${theme.colors.vegetarian}
  }
  .agilysys-icon-vegan {
    color: ${theme.colors.vegan}
  }
  .agilysys-icon-spicy {
    color: ${theme.colors.spicy}
  }
  .agilysys-icon-gluten-free {
    color: ${theme.colors.glutenFree}
  }
  @media print {
    @-moz-document url-prefix() {
      .appContainer, .appContainer>div, .appContainer .parent, .appContainer .BottomContainer {
        display: block !important;
      }
    }
  }
`;

export default theme;
