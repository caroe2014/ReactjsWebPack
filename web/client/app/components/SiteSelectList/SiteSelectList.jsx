// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React, { Component } from 'react';
import styled, { ThemeProvider } from 'styled-components';
import theme from 'web/client/theme';
import PropTypes from 'prop-types';
import { Flex } from 'rebass';
import SiteSelectCard from 'web/client/app/components/SiteSelectCard';
import { getOrderConfigurationDetailsWithContextId } from 'web/client/app/utils/OrderConfig';
import i18n from 'web/client/i18n';

const Container = styled(Flex)`
  flex-direction: column;
  margin: 70px auto 0px auto;
  height: fit-content;
  max-width: 75em;
  width: 100%;
  padding: 0px;
  &:after{
    content: '';
    margin: auto;
    width: 100%;
    max-width: 570px;
    }
  ${props => props.theme.mediaBreakpoints.mobile`
    justify-content: center;
    padding-left: 10px;
    padding-right: 10px;
  `}
  ${props => props.theme.mediaBreakpoints.tablet`
    justify-content: center;
  `}
  ${props => props.theme.mediaBreakpoints.desktop`
    justify-content: space-between;
    padding-left: 20px;
    padding-right: 20px;
  `}
`;

const ListContainer = styled(Flex)`
  flex-direction: column;
  flex-flow: row wrap;
  height: fit-content;
  max-width: 75em;
  margin-top: 80px;
  width: 100%;
  padding: 0px;
  &:after{
    content: '';
    margin: auto;
    width: 100%;
    max-width: 570px;
    }
  ${props => props.theme.mediaBreakpoints.mobile`
    justify-content: center;
  `}
  ${props => props.theme.mediaBreakpoints.tablet`
    justify-content: center;
  `}
  ${props => props.theme.mediaBreakpoints.desktop`
    justify-content: space-between;
  `}
  &:focus {
    outline: none;
    box-shadow: none;
  }
`;

class SiteSelectList extends Component {
  constructor (props) {
    super(props);
    this.selectSite = this.selectSite.bind(this);
    this.filterStoresWithAsapDisabled = this.filterStoresWithAsapDisabled.bind(this);
  }

  componentDidMount () {
    const { keyProps, skipSitesPage } = this.props;

    if (skipSitesPage) {
      this.selectSite(keyProps[0]);
    } else if (!skipSitesPage && document.getElementById('top-container')) {
      document.getElementById('top-container').focus();
    }
  }

  selectSite (site) {
    const { selectSite, setOrderConfig, keyProps } = this.props;
    if (selectSite) {
      const orderConfig = getOrderConfigurationDetailsWithContextId(site.id, site.displayProfileId, keyProps);
      setOrderConfig(orderConfig);
      selectSite(site);
    }
  }

  filterStoresWithAsapDisabled (site) {
    const { isAsapOrder } = this.props;
    return !isAsapOrder || !site.isAsapOrderDisabled;
  }

  render () {
    const { keyProps, apptheme, scheduleOrderEnabled, showOperationTimes, setDeliveryOption,
      isAsapOrder } = this.props;
    return (
      <ThemeProvider className='site-list' theme={theme}>
        <Container className='site-list-container' p={[1, 2, 3, 4]} m={[0, 2]}>
          <ListContainer className='site-list-container'
            tabIndex={0}
            aria-label={keyProps.length > 1 ? `${i18n.t('STORE_SELECTION')} ${keyProps.length} ${i18n.t('STORES_LABEL')}` : ``}> {/* eslint-disable-line max-len */}
            {keyProps && keyProps.filter(this.filterStoresWithAsapDisabled).map((item, index) => (
              <React.Fragment key={`Site-${item.id}-${item.displayProfileId}`}>
                <SiteSelectCard
                  className={`tile site-${index + 1}`}
                  keyProps={item}
                  selectSite={this.selectSite}
                  apptheme={apptheme}
                  isScheduleOrderEnabled={scheduleOrderEnabled}
                  isAsapOrder={isAsapOrder}
                  showOperationTimes={showOperationTimes}
                  setDeliveryOption={setDeliveryOption}
                  ariaLabel={`store${index + 1} ${item.name}`}
                  tabIndex={0}
                />
              </React.Fragment>
            ))}
          </ListContainer>
        </Container>
      </ThemeProvider>
    );
  }
}

SiteSelectList.propTypes = {
  selectSite: PropTypes.func,
  keyProps: PropTypes.arrayOf(
    PropTypes.shape({
      image: PropTypes.string,
      name: PropTypes.string.isRequired,
      availableAt: PropTypes.shape({
        opens: PropTypes.string.isRequired,
        closes: PropTypes.string.isRequired
      }).isRequired,
      address: PropTypes.arrayOf(PropTypes.string.isRequired)
    }).isRequired
  )
};

export default SiteSelectList;
