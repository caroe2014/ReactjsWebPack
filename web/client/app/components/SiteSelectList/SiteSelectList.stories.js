// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React from 'react';
import { storiesOf } from '@storybook/react';
import SiteSelectList from './SiteSelectList';

storiesOf('SiteSelectList', module)
  .add('With Images', () => {
    const siteItems = [{
      image: 'http://eatstreet.co.id/wp-content/uploads/2016/11/cropped-MASTER-FINAL_LOGO-OVAL-EAT-STREET-1.png',
      name: 'Eat Street',
      availableAt: {
        opens: '9:00am',
        closes: '6:00pm'
      },
      address: ['1000, windward concourse', 'alpharetta,GA, USA']
    },
    {
      image: 'https://i.ebayimg.com/images/g/CXoAAOSwo4pYHIIl/s-l300.jpg',
      name: 'City Central',
      availableAt: {
        opens: '9:00am',
        closes: '5:00pm'
      },
      address: ['3380-146th PI SE', 'Believue,USA']
    },
    {
      image: 'http://www.eamc.in/wp-content/uploads/2015/12/inorbit-logo-2.png',
      name: 'Inorbit',
      availableAt: {
        opens: '9:00am',
        closes: '5:00pm'
      },
      address: ['6775, S Edmond st #100', 'Lasvegas, USA']
    },
    {
      image: 'https://upload.wikimedia.org/wikipedia/commons/7/74/Mall_of_america_logo13.png',
      name: 'Mall of America',
      availableAt: {
        opens: '9:00am',
        closes: '5:00pm'
      },
      address: ['Santa Barbara', 'CA, USA']
    },
    {
      image: 'http://eatstreet.co.id/wp-content/uploads/2016/11/cropped-MASTER-FINAL_LOGO-OVAL-EAT-STREET-1.png',
      name: 'Eat Street',
      availableAt: {
        opens: '9:00am',
        closes: '5:00pm'
      },
      address: ['1000, windward concourse', 'alpharetta,GA, USA']
    },
    {
      image: 'https://i.ebayimg.com/images/g/CXoAAOSwo4pYHIIl/s-l300.jpg',
      name: 'City Central',
      availableAt: {
        opens: '9:00am',
        closes: '5:00pm'
      },
      address: ['3380-146th PI SE', 'Believue,USA']
    },
    {
      image: 'http://www.eamc.in/wp-content/uploads/2015/12/inorbit-logo-2.png',
      name: 'Inorbit',
      availableAt: {
        opens: '9:00am',
        closes: '2:00pm'
      },
      address: ['6775, S Edmond st #100', 'Lasvegas, USA']
    }];
    return (
      <SiteSelectList keyProps={siteItems} />
    );
  })
  .add('Without Images', () => {
    const siteItems = [{
      name: 'Eat Street',
      availableAt: {
        opens: '9:00am',
        closes: '5:00pm'
      },
      address: ['1000, windward concourse', 'alpharetta,GA, USA']
    },
    {
      name: 'City Central',
      availableAt: {
        opens: '9:00am',
        closes: '5:00pm'
      },
      address: ['3380-146th PI SE', 'Believue,USA']
    },
    {
      name: 'Inorbit',
      availableAt: {
        opens: '9:00am',
        closes: '5:00pm'
      },
      address: ['6775, S Edmond st #100', 'Lasvegas, USA']
    },
    {
      name: 'Mall of America',
      availableAt: {
        opens: '9:00am',
        closes: '5:00pm'
      },
      address: ['Santa Barbara', 'CA, USA']
    },
    {
      name: 'Eat Street',
      availableAt: {
        opens: '9:00am',
        closes: '5:00pm'
      },
      address: ['1000, windward concourse', 'alpharetta,GA, USA']
    },
    {
      name: 'City Central',
      availableAt: {
        opens: '9:00am',
        closes: '5:00pm'
      },
      address: ['3380-146th PI SE', 'Believue,USA']
    },
    {
      name: 'Inorbit',
      availableAt: {
        opens: '9:00am',
        closes: '2:00pm'
      },
      address: ['6775, S Edmond st #100', 'Lasvegas, USA']
    },
    {
      name: 'Mall of America',
      availableAt: {
        opens: '9:00am',
        closes: '5:00pm'
      },
      address: ['Santa Barbara', 'CA, USA']
    }];
    return (
      <SiteSelectList keyProps={siteItems} />
    );
  });
