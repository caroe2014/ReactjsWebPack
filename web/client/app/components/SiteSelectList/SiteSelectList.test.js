// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { ThemeProvider } from 'styled-components';
import theme from 'web/client/theme.js';
import SiteSelectList from './SiteSelectList';
import { t } from 'i18next';

configure({ adapter: new Adapter() });

jest.mock('i18next');
t.mockImplementation((key) => {
  return key;
});

jest.mock('react-i18next', () => ({
  Trans: 'div'
}));

jest.mock('web/client/i18n', () => ({
  t: (key) => {
    return key;
  }
}));

describe('SiteSelectList', () => {

  let wrapper;
  let mockProps = [
    {
      id: '1000',
      image: 'image.png',
      name: 'This is a mock name',
      availableAt: {
        opens: '9:00am',
        closes: '5:00pm',
        availableNow: true,
        closingIn: 10
      },
      address: ['1000, windward concourse', 'alpharetta,GA, USA']
    },
    {
      id: '1001',
      image: 'image.png',
      name: 'This is a mock name',
      availableAt: {
        opens: '9:00am',
        closes: '5:00pm',
        availableNow: true,
        closingIn: 10
      },
      address: ['3380-146th PI SE', 'Believue,USA']
    },
    {
      id: '1002',
      image: 'image.png',
      name: 'This is a mock name',
      availableAt: {
        opens: '9:00am',
        closes: '5:00pm',
        availableNow: true,
        closingIn: 10
      },
      address: ['6775, S Edmond st #100', 'Lasvegas, USA']
    }
  ];

  beforeEach(() => {
    wrapper = shallow(<SiteSelectList keyProps={mockProps} />
    );
  });

  it('does a snapshot test', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render without throwing an error', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('should render without throwing an error in mount', () => {
    const mountWrapper = mount(<ThemeProvider theme={theme}><SiteSelectList keyProps={mockProps} /></ThemeProvider>);
    expect(mountWrapper.exists()).toBe(true);
  });

  describe('when Site list props are received', () => {
    it('should render the same number of tiles', () => {
      expect(wrapper.find('.tile').length).toBe(mockProps.length);
    });
  });

  describe('when Site list props are not received', () => {
    beforeEach(() => {
      mockProps = [];
    });
    it('should not render ItemComponent', () => {
      const itemWrapper = shallow(<SiteSelectList keyProps={mockProps} />);
      expect(itemWrapper.find('.tile').length).toBe(0);
    });
  });

  it('should render without throwing', () => {
    const mountWrapper = shallow(<SiteSelectList keyProps={mockProps} selectSite/>);
    wrapper.instance().forceUpdate();
    wrapper.instance().selectSite();
    expect(mountWrapper.exists()).toBe(true);
  });

});
