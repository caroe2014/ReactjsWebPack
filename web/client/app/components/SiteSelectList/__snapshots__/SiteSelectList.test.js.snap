// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteSelectList does a snapshot test 1`] = `
<ThemeProvider
  className="site-list"
  theme={
    Object {
      "colors": Object {
        "addOnTab": "#347CFF",
        "addToCart": "#347CFF",
        "addonBg": "#EEEEEE",
        "backgroundGrey": "#E6E6E6",
        "black": "#000000",
        "close": "red",
        "contextBackground": "#f0f0f3",
        "darkGrey": "#464646",
        "disableBackground": "#E4E4E4",
        "disableButton": "#D1D1D1",
        "error": "#cc0000",
        "footer": "#48a942",
        "glutenFree": "#8d7356",
        "healthy": "#359919",
        "hint": "#a6a6a6",
        "inputBottomBorder": "lightgray",
        "itemBorder": "lightgrey",
        "itemTile": "#3C474C0",
        "light": "white",
        "loader": "#ababab",
        "loginBackground": "#303030",
        "loginBoxColor": "#434149",
        "loginField": "#AAA7AC",
        "loginFormBackground": "#484848",
        "navigationMenu": "#48a942",
        "navigationMenuHover": "cornflowerblue",
        "new": "#ff6000",
        "open": "#48a942",
        "paymentSuccessTitle": "#505050",
        "placeHolder": "#909090",
        "primary": "#347CFF",
        "primaryButton": "#347CFF",
        "primaryText": "#383838",
        "secondaryText": "#a6a6a6",
        "spicy": "#ed1c24",
        "text": "black",
        "textGrey": "#505050",
        "tile": "#505050",
        "validationError": "#cc0000",
        "vegan": "#0d4a22",
        "vegetarian": "#0d4a22",
      },
      "fontFamily": "Roboto",
      "fontSize": Object {
        "lg": "1.8rem",
        "md": "1.3rem",
        "nm": "1rem",
        "sm": "0.75rem",
        "xl": "2.4rem",
      },
      "fontSizeFromProps": [Function],
      "mediaBreakpoints": Object {
        "desktop": [Function],
        "mobile": [Function],
        "tablet": [Function],
      },
      "nav": Object {
        "height": "70px",
      },
      "receiptFontFamily": "Roboto",
      "shadow": "#cfcfcf 0.1px 0.1px 3px 0px",
      "spacing": Object {
        "lg": "4em",
        "md": "2em",
        "nm": "1em",
        "none": "0em",
        "sm": "0.5em",
        "xl": "8em",
      },
    }
  }
>
  <Styled(Flex)
    className="site-list-container"
    m={
      Array [
        0,
        2,
      ]
    }
    p={
      Array [
        1,
        2,
        3,
        4,
      ]
    }
  >
    <Styled(Flex)
      aria-label="STORE_SELECTION 3 STORES_LABEL"
      className="site-list-container"
      tabIndex={0}
    >
       
      <SiteSelectCard
        ariaLabel="store1 This is a mock name"
        className="tile site-1"
        keyProps={
          Object {
            "address": Array [
              "1000, windward concourse",
              "alpharetta,GA, USA",
            ],
            "availableAt": Object {
              "availableNow": true,
              "closes": "5:00pm",
              "closingIn": 10,
              "opens": "9:00am",
            },
            "id": "1000",
            "image": "image.png",
            "name": "This is a mock name",
          }
        }
        selectSite={[Function]}
        tabIndex={0}
      />
      <SiteSelectCard
        ariaLabel="store2 This is a mock name"
        className="tile site-2"
        keyProps={
          Object {
            "address": Array [
              "3380-146th PI SE",
              "Believue,USA",
            ],
            "availableAt": Object {
              "availableNow": true,
              "closes": "5:00pm",
              "closingIn": 10,
              "opens": "9:00am",
            },
            "id": "1001",
            "image": "image.png",
            "name": "This is a mock name",
          }
        }
        selectSite={[Function]}
        tabIndex={0}
      />
      <SiteSelectCard
        ariaLabel="store3 This is a mock name"
        className="tile site-3"
        keyProps={
          Object {
            "address": Array [
              "6775, S Edmond st #100",
              "Lasvegas, USA",
            ],
            "availableAt": Object {
              "availableNow": true,
              "closes": "5:00pm",
              "closingIn": 10,
              "opens": "9:00am",
            },
            "id": "1002",
            "image": "image.png",
            "name": "This is a mock name",
          }
        }
        selectSite={[Function]}
        tabIndex={0}
      />
    </Styled(Flex)>
  </Styled(Flex)>
</ThemeProvider>
`;
